lib/ansible/modules/system/java_cert.py

```diff
@@ -373,23 +357,27 @@ def main():
     cert_present = check_cert_present(module, executable, keystore_path,
                                       keystore_pass, cert_alias, keystore_type)
 
-    if state == 'absent':
-        if cert_present:
-            delete_cert(module, executable, keystore_path, keystore_pass, cert_alias, keystore_type)
+    if state == 'absent' and cert_present:
+        if module.check_mode:
+            module.exit_json(changed=True)
+
+        delete_cert(module, executable, keystore_path, keystore_pass, cert_alias, keystore_type)
+
+    elif state == 'present' and not cert_present:
+        if module.check_mode:
+            module.exit_json(changed=True)
 
-    elif state == 'present':
-        if not cert_present:
-            if pkcs12_path:
-                import_pkcs12_path(module, executable, pkcs12_path, keystore_path,
-                                   keystore_pass, pkcs12_pass, pkcs12_alias, cert_alias, keystore_type)
+        if pkcs12_path:
+            import_pkcs12_path(module, executable, pkcs12_path, keystore_path,
+                               keystore_pass, pkcs12_pass, pkcs12_alias, cert_alias, keystore_type)
 
-            if path:
```

Comment:
Not really a regression, but these tree if blocks are not supposed to be called in a row. So it may be interesting to:
- ensure the tree parameters are not set all together
- use `elif` statement instead of the `if`